pandas/tests/io/parser/test_index_col.py

```diff
@@ -297,3 +297,24 @@ def test_multiindex_columns_index_col_with_data(all_parsers):
         index=Index(["data"]),
     )
     tm.assert_frame_equal(result, expected)
+
+
+@skip_pyarrow
+def test_infer_types_boolean_sum(all_parsers):
+    # GH#44079
+    parser = all_parsers
+    result = parser.read_csv(
+        StringIO("0,1"),
+        names=["a", "b"],
+        index_col=["a"],
+        dtype={"a": "UInt8"},
+    )
+    expected = DataFrame(
+        data={
+            "a": [
+                0,
+            ],
+            "b": [1],
+        }
+    ).set_index("a")
+    tm.assert_frame_equal(result, expected, check_index_type=False)
```

Comment:
why is the check_index_type=False? can we not construct the appropriate expected?

Reply:
The C parser returns a index column of dtype 'object' and the Python parser returns a index column of dtype 'int64'. Both result are inferred, and both make sense in this specific example. Unifying the behavior of the parsers is desirable but beyond the scope of the current bug fix.

Reply:
can you add a comment about why this is not checking index type (basically what you said here)

Reply:
done